code-challenges.test.js

```diff
@@ -36,23 +71,84 @@ const filterLetterA = "a"
 const filterLetterE = "e"
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
+// describe("filteredFruit", () => {
+//     it("returns array filtered fruit", () => {
+//         expect(filteredFruit(fruitArray, filterLetterA)).toEqual(["Mango", "Apricot", "Peach"])
+//         expect(filteredFruit(fruitArray, filterLetterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+//     })
+// })
+
 // b) Create the function that makes the test pass.
 
-// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
+const filteredFruit = (fruitArr, char) => {
+    return fruitArr.filter(fruit => fruit.toLowerCase().includes(char))
+}
+
+// console.log(filteredFruit(fruitArray, filterLetterA))
+// console.log(filteredFruit(fruitArray, filterLetterE))
 
 // Pseudo code:
 
+// Define a function that takes an array of fruits and character as arguments
+// Use the filter() to create new array of filtering out the fruits that include the inputed character
+// Return filtered array 
+
+// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
+
 // a) Create a test with expect statements using the variable provided.
 
+describe("winningHand", () =>{
+    it("returns true if these id full house or false.", () =>{
+        expect(winningHand(hand1)).toEqual(true)
+        expect(winningHand(hand2)).toEqual(false)
+        expect(winningHand(hand3)).toEqual(false)
+        expect(winningHand(hand4)).toEqual(true)
+    })
+})
+
 const hand1 = [5, 5, 5, 3, 3]
 // Expected output: true
+
 const hand2 = [5, 5, 3, 3, 4]
 // Expected output: false
+
 const hand3 = [5, 5, 5, 5, 4]
+
 // Expected output: false
+
 const hand4 = [7, 2, 7, 2, 7]
+
 // Expected output: true
 
 // b) Create the function that makes the test pass.
 
+const winningHand = (arr) => {
+    const num_counts = {}
+    for (num of arr){
+        num_counts[num] = (num_counts[num] || 0) + 1
+    }
+    const values = Object.values(num_counts);
+    return values.includes(2) && values.includes(3)
+}
+console.log(winningHand(hand1))
+console.log(winningHand(hand2))
+console.log(winningHand(hand2))
+console.log(winningHand(hand4))
+
 // Pseudo code:
+// create function that accepts array and returns true or false
+//      create an empty dictionary to store the counts of numbers
+
+//      iterate through each num in arr
+//         IF num_counts[num] EXISTS
+//             num_counts[num] = num_counts[num] + 1
+//         ELSE
+//             num_counts[num] = 1
+
+//      values = get values from num_counts
+
+//      IF values contain 2 AND values contain 3
+//         return True
+//      ELSE
+//         return False
```

Comment:
Great logic for this one, nice job figuring this out. I suggest making your pseudo code less code-like and more words. The point is to quickly explain what the code is doing.